js/models/base.js

```diff
@@ -94,15 +94,16 @@ export class Base {
      *
      * @param  {String}   endpoint The API endpoint to call
      * @param  {Object}   data     The data object to submit
-     * @param  {Function} callback The callback function to call on success.
+     * @param  {Function} on_success The callback function to call on success.
+     * @param  {Function} on_error The callback function to call on error.
      */
-    $api(endpoint, data, callback) {
-        var parts = endpoint.split('.'),
-            namespace = parts[0],
-            method = parts[1],
-            operation = API[namespace][method];
+    $api(endpoint, data, on_success, on_error=()=>{}) {
```

Comment:
I think it will introduce multiple binding and then memory leak.
May be better to have ``$api(endpoint, data, on_success, on_error)``
and then ``(on_error || () =>{}).bind(this)``